src/Sylius/Component/Core/spec/OrderProcessing/OrderPaymentProcessorSpec.php

```diff
@@ -46,13 +47,21 @@ function it_throws_exception_if_passed_order_is_not_core_order(BaseOrderInterfac
 
     function it_removes_cart_payments_from_order_if_its_total_is_zero(
         OrderInterface $order,
-        PaymentInterface $cartPayment
+        PaymentInterface $cartPayment,
+        PaymentInterface $cancelledPayment,
+        Collection $payments
     ): void {
         $order->getState()->willReturn(OrderInterface::STATE_CART);
         $order->getTotal()->willReturn(0);
 
-        $order->getPayments(OrderPaymentStates::STATE_CART)->willReturn(new ArrayCollection([$cartPayment->getWrappedObject()]));
-        $order->removePayment($cartPayment)->shouldBeCalled();
+        $cartPayment->getState()->willReturn(OrderPaymentStates::STATE_CART);
+        $cancelledPayment->getState()->willReturn(OrderPaymentStates::STATE_CANCELLED);
+
+        $payments->filter(Argument::type(\Closure::class))->willReturn([$cartPayment])->shouldBeCalledTimes(1);
```

Comment:
We don't need `shouldBeCalled...` if we have `willReturn`

Reply:
👍 done